### PR TITLE
[WishlistBundle] new feature: wishlist cleaner

### PR DESCRIPTION
**Pull request by alexloetscher95**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no

Adds the posibility to clean up old wishlists similar to the CartExpireTask

Similar to the CartExpireTask, the expiration is deactivated by default and can be activated by following configuration (example): 
```yml
core_shop_order:
    expiration:
        wishlist:
            days: 20
            anonymous: true
            customer: false
```

### CODE REVIEW

**Comment by dpfaffenbauer** [comment]
@alexloetscher95 you are happy so far with it right?

File: src/CoreShop/Bundle/WishlistBundle/Resources/config/pimcore/config.yml [hunk 1]
```diff
@@ -27,4 +27,14 @@ core_shop_storage_list:
...
                 summary: '@CoreShopFrontend/Wishlist/summary.html.twig'
             controller:
-                enabled: true
\ No newline at end of file
+                enabled: true
+            expiration:
+                enabled: true
+                service: CoreShop\Bundle\OrderBundle\Expiration\OrderAndCartExpiration
+                params:
+                    cart:
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/WishlistBundle/Resources/config/pimcore/config.yml:35]: doesn't make sense for wishlist

File: src/CoreShop/Bundle/CoreBundle/Pimcore/Repository/WishlistRepository.php [hunk 2]
```diff
@@ -20,11 +20,14 @@
...
 use CoreShop\Component\Core\Wishlist\Repository\WishlistRepositoryInterface;
 use CoreShop\Component\Customer\Model\CustomerInterface;
+use CoreShop\Component\StorageList\Repository\CustomerExpiryRepositoryTrait;
+use CoreShop\Component\StorageList\Repository\ExpireAbleStorageListRepositoryInterface;
 use CoreShop\Component\Store\Model\StoreInterface;
 use CoreShop\Component\Wishlist\Model\WishlistInterface;
-
-class WishlistRepository extends \CoreShop\Bundle\WishlistBundle\Pimcore\Repository\WishlistRepository implements WishlistRepositoryInterface
+use CoreShop\Bundle\WishlistBundle\Pimcore\Repository\WishlistRepository as BaseWishlistRepository;
+class WishlistRepository extends BaseWishlistRepository implements WishlistRepositoryInterface, ExpireAbleStorageListRepositoryInterface
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/CoreBundle/Pimcore/Repository/WishlistRepository.php:28]: newline between use and class

**Comment by dpfaffenbauer** [comment]
@alexloetscher95 nice, I will check the PR again this week and then merge it

**Review by dpfaffenbauer — APPROVED** [review]

**Comment by dpfaffenbauer** [comment]
@alexloetscher95 thanks a lot for kicking this off and helping with it :)

**Comment by aarongerig** [comment]
Any chance to get a release for this feature? :)

**Comment by dpfaffenbauer** [comment]
will be part of 3.1

**Comment by aarongerig** [comment]
Any ETA? Just so I know, if I have to come up with another plan... 😉

**Comment by dpfaffenbauer** [comment]
about 1-2 weeks

**Pull request merged**